prompt2model/dataset_generator/prompt_based.py

```diff
@@ -290,35 +221,20 @@ def apply_multi_vote_to_construct_generated_dataset(
         highest frequency, it selects the one that comes first in lexicographical
         (alphabetical) order.
 
-        Example:
-        Suppose input_output_map is:
-        {
-            "apple": Counter({"A": 2, "D": 2}),
-            "banana": Counter({"B": 2, "C": 1}),
-            "orange": Counter({"O": 1})
-        }
-
-        The function will produce generated_dataset:
-        {
-            "input_col": ["apple", "banana", "orange"],
-            "output_col": ["A", "B", "O"]
-        }
-
-        Note: When generated_examples is empty, both input_output_map
-        and generated_dataset will be empty.
-
         Returns:
             Currently generated dataset with multi-vote filtering applied.
         """
-        # Ensure that multi-vote filtering is enabled.
-        if not self.filter_duplicated_examples:
-            raise ValueError("Multi-vote filtering is not enabled.")
+        filtered_examples = []
+
+        input_output_map: dict[str, Counter] = defaultdict(Counter)
 
-        filtered_inputs = []
-        filtered_outputs = []
+        for ex in generated_examples:
+            input_output_map[ex.input_col][ex.output_col] += 1
+
+        if len(generated_examples) != 0 and input_output_map is None:
+            raise ValueError("input_output_map is not correctly constructed.")
```

Comment:
Will this ever be satisfied? Since `input_output_map` is initialized as
```
input_output_map: dict[str, Counter] = defaultdict(Counter)
```

I'm pretty sure it'll _never_ be None.